crates/iceberg/src/table.rs

```diff
@@ -203,7 +203,7 @@ impl Table {
 
     /// Creates a metadata table which provides table-like APIs for inspecting metadata.
     /// See [`MetadataTable`] for more details.
-    pub fn metadata_table(self) -> MetadataTable {
+    pub fn metadata_table(&self) -> MetadataTable<'_> {
```

Comment:
Follow-up to this comment: https://github.com/apache/iceberg-rust/pull/822#discussion_r1899617821

Also not necessary to do in this PR.